packages/askar/src/wallet/AskarWallet.ts

```diff
@@ -0,0 +1,745 @@
+    }
+
+    try {
+      await this.session.close()
+      await this.store.close()
+      this._session = undefined
+      this._store = undefined
+      this.publicDidInfo = undefined
+    } catch (error) {
+      const errorMessage = `Error closing wallet': ${error.message}`
+      this.logger.error(errorMessage, {
+        error,
+        errorMessage: error.message,
+      })
+
+      throw new WalletError(errorMessage, { cause: error })
+    }
+  }
+
+  public async initPublicDid(didConfig: DidConfig) {
+    // Not implemented, as it does not work with legacy Ledger module
+  }
+
+  /**
+   * Create a key with an optional seed and keyType.
+   * The keypair is also automatically stored in the wallet afterwards
+   *
+   * @param seed string The seed for creating a key
+   * @param keyType KeyType the type of key that should be created
+   *
+   * @returns a Key instance with a publicKeyBase58
+   *
+   * @throws {WalletError} When an unsupported keytype is requested
+   * @throws {WalletError} When the key could not be created
+   */
+  public async createKey({ seed, keyType }: WalletCreateKeyOptions): Promise<Key> {
+    try {
+      if (keyTypeSupportedByAskar(keyType)) {
+        const algorithm = keyAlgFromString(keyType)
+
+        // Create key from seed
+        const key = seed
+          ? AskarKey.fromSeed({ seed: new TextEncoder().encode(seed), algorithm })
+          : AskarKey.generate(algorithm)
+
+        // Store key
+        await this.session.insertKey({ key, name: encodeToBase58(key.publicBytes) })
+        return Key.fromPublicKey(key.publicBytes, keyType)
+      } else {
+        // Check if there is a signing key provider for the specified key type.
+        if (this.signingKeyProviderRegistry.hasProviderForKeyType(keyType)) {
+          const signingKeyProvider = this.signingKeyProviderRegistry.getProviderForKeyType(keyType)
+
+          const keyPair = await signingKeyProvider.createKeyPair({ seed })
+          await this.storeKeyPair(keyPair)
+          return Key.fromPublicKeyBase58(keyPair.publicKeyBase58, keyType)
+        }
+        throw new WalletError(`Unsupported key type: '${keyType}'`)
+      }
+    } catch (error) {
+      if (!isError(error)) {
+        throw new AriesFrameworkError('Attempted to throw error, but it was not of type Error', { cause: error })
+      }
+      throw new WalletError(`Error creating key with key type '${keyType}': ${error.message}`, { cause: error })
+    }
+  }
+
+  /**
+   * sign a Buffer with an instance of a Key class
+   *
+   * @param data Buffer The data that needs to be signed
+   * @param key Key The key that is used to sign the data
+   *
+   * @returns A signature for the data
+   */
+  public async sign({ data, key }: WalletSignOptions): Promise<Buffer> {
+    try {
+      if (!TypedArrayEncoder.isTypedArray(data)) {
+        throw new WalletError(`Currently not supporting signing of multiple messages`)
+      }
```

Comment:
The BBS Signing provider does support multilpe message I think? Maybe we can place this withiin the `if (keyTypeSupportedByAskar(key.keyType)) {`?